runtime/service.go

```diff
@@ -1182,7 +1183,7 @@ func (s *service) Shutdown(requestCtx context.Context, req *taskAPI.ShutdownRequ
 
 	err = s.machine.StopVMM()
 	if err != nil {
-		shutdownErr = multierror.Append(shutdownErr, errors.Wrap(err, "failed to gracefully stop VM"))
+		shutdownErr = multierror.Append(shutdownErr, errors.Wrap(err, "failed to stop VM"))
 	}
```

Comment:
It's good to know that `StopVMM` is not implemented as "graceful" shutdown, but can we open a followup issue to update this to actually be graceful (i.e. send `CtlAltDel` and wait a bit)? Based on [this issue](https://github.com/firecracker-microvm/firecracker/issues/1370) it seems like there are risks associated with just doing a `SIGTERM` right away. We'll need to update the vm guest kernel used by our CI system's I'm guessing, but that probably worth the effort here.

Reply:
I didn't know the issue. Thanks!

Regarding SendCtrlAltDel, it doesn't work with the kernel we are using, it has been [disabled on ARM](https://github.com/firecracker-microvm/firecracker/pull/1341)since [there is no i8042 device](https://github.com/firecracker-microvm/firecracker/issues/1339).

How about rebooting a guest from our agent? It is slightly complicated from SendCtrlAltDel, but it doesn't have the issues I've mentioned.

Reply:
Oh +100 to doing the reboot from Agent, I like that idea a lot more